src/esp/assets/ResourceManager.cpp

```diff
@@ -1811,41 +1813,106 @@ void ResourceManager::loadTextures(Importer& importer,
-    // be loaded by now
-    if (resourceDict_.count(renderObjectName) == 0) {
+  // get render asset handle
+  std::string renderAssetHandle =
+      physicsObjectAttributes->getRenderAssetHandle();
+  // whether attributes requires lighting
+  bool requiresLighting = physicsObjectAttributes->getRequiresLighting();
+  bool renderMeshSuccess = false;
+  // no resource dict entry exists for renderAssetHandle
+  if (resourceDict_.count(renderAssetHandle) == 0) {
+    if (physicsObjectAttributes->getRenderAssetIsPrimitive()) {
       // needs to have a primitive asset attributes with same name
-      if (primitiveAssetsTemplateLibrary_.count(renderObjectName) == 0) {
+      if (primitiveAssetsTemplateLibrary_.count(renderAssetHandle) == 0) {
         // this is bad, means no render primitive template exists with expected
         // name.  should never happen
         LOG(ERROR) << "No primitive asset attributes exists with name :"
-                   << renderObjectName
+                   << renderAssetHandle
                    << " so unable to instantiate primitive-based render "
                       "object.  Aborting.";
-        return;
+        return false;
       }
       // build primitive asset for this object based on defined primitive
       // attributes
       auto primitiveAssetAttributes =
-          primitiveAssetsTemplateLibrary_.at(renderObjectName);
+          primitiveAssetsTemplateLibrary_.at(renderAssetHandle);
       buildPrimitiveAssetData(primitiveAssetAttributes);
+
+    } else {
+      // load/check_for render mesh metadata and load assets
+      renderMeshSuccess = loadObjectMeshDataFromFile(
+          renderAssetHandle, objectTemplateHandle, "render", requiresLighting);
+    }
+  }  // if no render asset exists
+
+  // check if uses collision mesh
+  if (!physicsObjectAttributes->getCollisionAssetIsPrimitive()) {
+    const auto collisionAssetHandle =
+        physicsObjectAttributes->getCollisionAssetHandle();
+    if (resourceDict_.count(collisionAssetHandle) == 0) {
+      bool collisionMeshSuccess = loadObjectMeshDataFromFile(
+          collisionAssetHandle, objectTemplateHandle, "collision",
+          !renderMeshSuccess && requiresLighting);
+
+      if (!collisionMeshSuccess) {
+        return false;
+      }
     }
+    // check if collision handle exists in collision mesh groups yet.  if not
+    // then instance
+    if (collisionMeshGroups_.count(collisionAssetHandle) == 0) {
+      // set collision mesh data
+      const MeshMetaData& meshMetaData = getMeshMetaData(collisionAssetHandle);
+
+      int start = meshMetaData.meshIndex.first;
+      int end = meshMetaData.meshIndex.second;
+      //! Gather mesh components for meshGroup data
+      std::vector<CollisionMeshData> meshGroup;
+      for (int mesh_i = start; mesh_i <= end; ++mesh_i) {
+        GenericMeshData* gltfMeshData =
+            dynamic_cast<GenericMeshData*>(meshes_[mesh_i].get());
```

Comment:
```suggestion
        GenericMeshData& gltfMeshData =
            dynamic_cast<GenericMeshData&>(*meshes_[mesh_i].get());
```

(I know it's not your code and it's just moved from elsewhere.) This will :boom: if the type doesn't match. With a pointer you'd get `nullptr` and then crash on that on the line below.